src/Product/WCProductAdapter.php

```diff
@@ -903,4 +907,149 @@ public function setTargetCountry( $targetCountry ) {
 		// product shipping information is also country based
 		$this->map_wc_product_shipping();
 	}
+
+	/**
+	 * Performs the attribute mapping.
+	 * This function applies rules setting values for the different attributes in the product.
+	 *
+	 * @param array $mapping_rules The set of rules to apply
+	 */
+	protected function map_attribute_mapping_rules( array $mapping_rules ) {
+		$attributes = [];
+
+		if ( empty( $mapping_rules ) ) {
+			return $this;
+		}
+
+		foreach ( $mapping_rules as $mapping_rule ) {
+			if ( $this->rule_match_conditions( $mapping_rule ) ) {
+				$attribute_id                = $mapping_rule['attribute'];
+				$attributes[ $attribute_id ] = self::format_attribute( apply_filters( "woocommerce_gla_product_attribute_value_{$attribute_id}", $this->get_source( $mapping_rule['source'] ), $this->get_wc_product() ), $attribute_id );
+			}
+		}
+
+		parent::mapTypes( $attributes );
+
+		// Size
+		if ( ! empty( $attributes['size'] ) ) {
+			$this->setSizes( [ $attributes['size'] ] );
+		}
+
+		return $this;
+	}
+
+	/**
+	 * Get a source value for attribute mapping
+	 *
+	 * @param string $source The source to get the value
+	 * @return string|null The source value for this product
+	 */
+	protected function get_source( string $source ) {
+		$source_type = null;
+
+		if ( strpos( $source, ':' ) ) {
+			$source_parts = explode( ':', $source );
+			$source_type  = $source_parts[0];
+			$source_value = $source_parts[1];
+		}
+
+		switch ( $source_type ) {
+			case 'product':
+				return self::get_product_field( $source_value );
+			case 'taxonomy':
+				return self::get_product_taxonomy( $source_value );
+			case 'attribute':
+				return $this->wc_product->get_meta( $source_value );
+			default:
+				return $source;
+		}
+	}
+
+	/**
+	 * Check if the current product match the conditions for applying the Attribute mapping rule.
+	 * For now the conditions are just matching with the product category conditions.
+	 *
+	 * @param array $rule The attribute mapping rule
+	 * @return bool True if the rule is applicable
+	 */
+	protected function rule_match_conditions( array $rule ): bool {
+		$attribute                 = $rule['attribute'];
+		$category_condition_type   = $rule['category_condition_type'];
+		$categories                = explode( ',', $rule['categories'] );
+		$wc_product                = $this->get_wc_product();
+		$terms                     = get_the_terms( $wc_product->get_id(), 'product_cat' );
+		$wc_product_categories     = $terms ? wp_list_pluck( $terms, 'term_id' ) : [];
```

Comment:
> The helper function get_wc_product is there for when we need to get it externally, but internally it just seems like overkill since we can just access $this->wc_product.

The potential issue of calling directly `$this->wc_product` is that if the getter had some kind of decoration (Let's say a filter.) We could have inconsistencies. Yes, it's a short chance of that happen. But as short is the "overkill" for using the getter.

Reply:
Makes sense, that part was just a minor comment. I just noticed that in most other places in that same class it just called the variable directly, so if we would add that filter then we would need to make it consistent everywhere.